addons/account_edi/models/account_edi_format.py

```diff
@@ -499,12 +499,21 @@ def _retrieve_partner(self, name=None, phone=None, mail=None, vat=None):
         :param vat:     The vat number of the partner.
         :returns:       A partner or an empty recordset if not found.
         '''
+        normalized_vat = vat
+        if vat and len(vat) > 2:
+            if vat[0].isalpha() and vat[1].isalpha():
+                normalized_vat = vat[2:]
+            try:
+                normalized_vat = str(int(normalized_vat))
+            except ValueError:
+                pass
+
         domains = []
         for value, domain in (
             (name, [('name', 'ilike', name)]),
             (phone, expression.OR([[('phone', '=', phone)], [('mobile', '=', phone)]])),
             (mail, [('email', '=', mail)]),
-            (vat, [('vat', 'like', vat)]),
+            (vat, expression.OR([[('vat', 'like', vat)], [('vat', 'like', normalized_vat)]])),
```

Comment:
'like' matches %pattern%, so here, assuming you have a VAT like BE00000123, you'll be matching anything that lis like %123%, so you could match BE123000000. I don't think you want that :p 

I'd rather user =like, here, and directly give it %normalized_vat. The first term sould probably do the same as well.